### ISSUE TITLE
[Feature] - Use browser notification API

### ISSUE THREAD

**Issue by gillg** [prymitive/karma#3328]
Being able to use the browser notif API would be very useful. It's not intrusive because you can choose to deny notifications, but it can help to keep an eye on Karma.
If in the same time we can customize  a sound it could be a great bonus.

**Comment by prymitive** [comment]
What kind of notifications are you thinking about?

**Comment by gillg** [comment]
I'm thinking about native browser / desktop notif. https://developer.mozilla.org/en-US/docs/Web/API/notification

As exemple with a test app https://mdn.github.io/to-do-notifications/
![image](https://user-images.githubusercontent.com/5639241/124937162-6d6a2d80-e007-11eb-9aca-5df8ca51ef53.png)

This is cross platform and is intergrated at OS level.

**Comment by prymitive** [comment]
Let me rephrase it: what message would you want to see in that notification?

**Comment by gillg** [comment]
Ah sorry ! 😅 
To brainstorm, but I think it needs to be configurable because all labels and annotations are arbitrary.

In my case for exemple, I imagine work at group level :
```
Notification.title = {{ $groupedAnnotations.summary }}
Notification.body = {{ $groupedAnnotations.description}}
```

An alternative :
```
Notification.title = "New Karma Alert !"
Notification.body = {{ $groupedAnnotations.summary }}
```

Notification.icon = can be hardcoded with a karma image, in a second time why not overridable
Notification.onclick = should open a karma URL equivalent to "Copy URL to this group"

We can also imagine work at alert level (if a new alert join an existing group and we want details), but I'm not sure if it's pertinent.
But we could use `$annotations` instead of `$groupedAnnotations`

**Comment by prymitive** [comment]
Here's the main problem with that - any decent modern browser will suspend inactive tab, so any updates karma runs will be paused if karma isn't the active tab. And if it's an active that you see all alerts hence you don't really benefit much from notifications.
To make it work karma would have to install a service worker that deals with background refreshes and calls notifications API. It's definitely doable but adds a completely new components, dependencies, configuration and functionality (none of which is trivial) and so maintenance cost. Plus you can just send alerts from alertmanager to slack/hipchat/msteams and have those notifications from it. All of that doesn't make it very likely to be added anytime soon. I might get at some point interested in adding a service worker, but it's nowhere near the top of my ever growing TODO list.

**Comment by gillg** [comment]
Hum, I understand your point.
I was thinking about a receiver like slack on alertmanager, but it depends on Karma usage. I would make it the first place to keep an eye on alerts (end even if it's not common... in my case usage of a third party chat is forbiden, so it make it more complex). Anyway I think it's a good feature to have, but I understand the complexity introduced by a background worker, I was not thinking about tab inactivity.
